### PR TITLE
Add FINOS Security Actions

### PR DESCRIPTION
**Pull request by TheJuanAndOnly99**
This PR adds 2 GitHub actions that perform a CVE and static code scan.

With the [CVE scanning](https://github.com/TheJuanAndOnly99/symphony-bdk-java/blob/security-scanning/.github/workflows/cve-scanning-gradle.yml) I could use some help as we are a bit unfamiliar with Gradle. The scan is passing but I am not sure if [dependencyCheckAnalyze task](https://github.com/TheJuanAndOnly99/symphony-bdk-java/actions/runs/3500333805/jobs/5862887362#step:5:29) is taking in consideration all project sub-modules. Could you please point me in the right direction here?

As far as the static code scan goes there were also many [findings](https://github.com/TheJuanAndOnly99/symphony-bdk-java/actions/runs/3497793441/jobs/5857329148). Please also take a look at these as we believe this tool does a good job at pointing out any potential vulnerabilities. If you need to bypass any you can do so (like I did [here](https://github.com/TheJuanAndOnly99/symphony-bdk-python/blob/security-scanning/examples/authentication/ext_app_auth.py#L17)).

Additional steps that would greatly improve the security posture of this project:

- Enable [RenovateBot](https://renovatebot.com/) - please let me know if I can enable it on this repo. This tool will help greatly with keeping your dependencies up to date.
- Add GitHub (security) Action badges at the top of the project README
- Apply for the OpenSSF Passing badge - https://bestpractices.coreinfrastructure.org/

Eager to hear your thoughts on it, and happy to adjust the aim based on your feedback.

### CODE REVIEW

**Comment by TheJuanAndOnly99** [comment]
Hi @yinan-symphony I added the Renovate bot I mentioned in the PR on my fork. You can see the results here https://github.com/TheJuanAndOnly99/symphony-bdk-java/pull/3.

File: .github/workflows/cve-scanning-gradle.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: CVE Scanning for Gradle
+
+on:
+  push:
+    paths:
```
Comment (symphony-soufiane) [inline .github/workflows/cve-scanning-gradle.yml:5]: I would prefer this action to run on every push/pull_request if it lasts a reasonable duration.

Comment (symphony-soufiane) [reply .github/workflows/cve-scanning-gradle.yml:5]: Also, the documentation mentions "dependencyCheckUpdate" task to update NVD from NIST. Do we need to add it to be sure the NVD is uptodate?

File: .github/workflows/cve-scanning-gradle.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+      - name: Build with Gradle
+        run: ./gradlew build
+      - name: CVEs
+        run: ./gradlew dependencyCheckAnalyze
```
Comment (symphony-soufiane) [inline .github/workflows/cve-scanning-gradle.yml:24]: I think that '_dependencyCheckAggregate_' should be used rather to scan all modules. 
Running dependencyCheckAnalyze showed in the report **_"Dependencies Scanned: 0 (0 unique)"_** while with '_dependencyCheckAggregate_' I can see **_'Dependencies Scanned: 151 (151 unique)'_**

Comment (yinan-symphony) [reply .github/workflows/cve-scanning-gradle.yml:24]: yes, in multi-module project, Aggregate should be the one to use.

**Comment by yinan-symphony** [comment]
> Hi @yinan-symphony I added the Renovate bot I mentioned in the PR on my fork. You can see the results here [TheJuanAndOnly99#3](https://github.com/TheJuanAndOnly99/symphony-bdk-java/pull/3).

@TheJuanAndOnly99 could you please configure the Renovate to set 
`{
  "dependencyDashboardApproval": true
}`
before enabling it in main branch.
many thx

File: .github/workflows/cve-scanning-gradle.yml (hunk 1 above)
Comment (TheJuanAndOnly99) [reply .github/workflows/cve-scanning-gradle.yml:5]: Hi @symphony-soufiane fair question. I think for a GitHub action it is not needed as the action does not cache any build artifacts. 

As for the action to run on every push/pull/request. It seems it takes around ~6 min to complete. I have added the change to the action if you want to go ahead with it.

File: .github/workflows/cve-scanning-gradle.yml (hunk 2 above)
Comment (TheJuanAndOnly99) [reply .github/workflows/cve-scanning-gradle.yml]: Thanks @symphony-soufiane  and @yinan-symphony! I've changed it in the PR. As you can see there are a [few CVEs.](https://github.com/TheJuanAndOnly99/symphony-bdk-java/actions/runs/3531749359/jobs/5925266182) Can you please take a look. If they are safe to ignore you can use the [allow-list.xml file](https://github.com/finos/security-scanning/blob/main/allow-list.xml) to do so.

**Comment by TheJuanAndOnly99** [comment]
@yinan-symphony I have added a renovate.json file to the PR with the `{ "dependencyDashboardApproval": true }` config.

**Comment by yinan-symphony** [comment]
@TheJuanAndOnly99 thank you for your contribute, could you please create a new PR to `feature/security-scanning` branch, so that i can update some config files before merging to main branch. Thank you in advance.

**Pull request merged**